python_modules/dagstermill/dagstermill/__init__.py

```diff
@@ -173,65 +173,79 @@ def serialize_dm_context(transform_execution_info, inputs):
     if not os.path.exists(marshal_dir):
         os.makedirs(marshal_dir)
 
-    new_inputs_structure = {
+    dm_context_dict = {
         'pipeline_name': transform_execution_info.pipeline_def.name,
         'solid_def_name': transform_execution_info.solid.definition.name,
-        'solid_name': transform_execution_info.solid.name,
-        'inputs': {},
         'marshal_dir': marshal_dir,
     }
 
+    parameters = dict(dm_context=json.dumps(dm_context_dict))
+
     input_defs = transform_execution_info.solid_def.input_defs
     input_def_dict = {inp.name: inp for inp in input_defs}
     for input_name, input_value in inputs.items():
+        assert (
+            input_name != "dm_context"
+        ), "Dagstermill solids cannot have inputs named 'dm_context'"
         runtime_type = input_def_dict[input_name].runtime_type
-
-        new_inputs_structure['inputs'][input_name] = marshal_value(
+        parameter_value = marshal_value(
             runtime_type, input_value, os.path.join(marshal_dir, 'input-{}'.format(input_name))
         )
-
-    return json.dumps(new_inputs_structure)
-
-
-def unmarshal_value(runtime_type, value):
-    check.inst_param(runtime_type, 'runtime_type', RuntimeType)
-    if runtime_type.is_scalar:
-        return value
-    elif runtime_type.is_any and is_json_serializable(value):
-        return value
-    elif runtime_type.marshalling_strategy:
-        return runtime_type.marshalling_strategy.unmarshal_value(value)
+        parameters[input_name] = parameter_value
+
+    return parameters
+
+
+def replace_parameters(info, nb, parameters):
```

Comment:
Can you write a comment saying how this is effectively a copy-paste job from papermill code and explain how you are creating a copy of the source notebook and removing the parameters cell?